spec/models/user_spec.rb

```diff
@@ -1,40 +1,32 @@
 require 'rails_helper'
-RSpec.describe UsersController, type: :model do
-  let(:user1) do
-    FactoryBot.build(:user, username: nil)
-  end
-  it 'is not valid without a first_name' do
-    expect(user1.valid?).to be_falsey
+RSpec.describe User, type: :model do
+  let(:user) do
+    FactoryBot.build(:user)
   end
 
-  let(:user2) do
-    FactoryBot.build(:user, password: nil)
-  end
-  it 'is not valid without a password' do
-    expect(user2.valid?).to be_falsey
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:username) }
+    it { is_expected.to validate_presence_of(:role) }
+    it { is_expected.to validate_presence_of(:password) }
+    it { is_expected.to validate_presence_of(:password_confirmation) }
   end
 
-  let(:user3) do
-    FactoryBot.build(:user, password_confirmation: nil)
-  end
-  it 'is not valid without a password_confirmation' do
-    expect(user3.valid?).to be_falsey
-  end
-
-  let(:user4) do
+  let(:user2) do
     FactoryBot.build(:user, password: '12', password_confirmation: '13')
   end
   it 'is not valid when password and password_confirmation not identical' do
-    expect(user4.valid?).to be_falsey
+    expect(user2.valid?).to be_falsey
   end
 
   let!(:user5) do
     FactoryBot.create(:user, username: 'test')
   end
   let(:user6) do
-    FactoryBot.build(:user, username: 'test')
+    FactoryBot.create(:user, username: 'test')
   end
-  it 'is not valid when naming 2 users with same username' do
-    expect(user6.save).to be_falsey
+  it { is_expected.to validate_uniqueness_of(:username) }
```

Comment:
Group all the let statements together at the top.
Move this validate_uniqueness_of inside the validations describe block.